lib/sqlitex/row.ex

```diff
@@ -56,8 +56,10 @@ defmodule Sqlitex.Row do
     {String.to_integer(yr), String.to_integer(mo), String.to_integer(da)}
   end
 
-  defp to_time(time) do
-    <<hr::binary-size(2), ":", mi::binary-size(2), ":", se::binary-size(2), ".", fr::binary-size(6)>> = time
+  defp to_time(<<hr::binary-size(2), ":", mi::binary-size(2), ":", se::binary-size(2), ".", fr::binary-size(6)>>) do
```

Comment:
Of course.  Good catch.  What about:

```
String.to_integer(fr <> String.duplicate("0", 6 - String.length(fr)))
```

To ensure we always return microseconds regardless of size.  That would require a guard on the function though to make sure we don't go negative:

```
def to_time(... fr::binary>>) when byte_size(fr) <= 6 do
```